docs/lib/datastore/fragments/flutter/relational/query-snippet.md

```diff
@@ -1,3 +1,8 @@
 ```dart
-TBD
+try {
+  List<Comment> comments = await Amplify.DataStore.query(Comment.classType,
+      where: Post.STATUS.eq(PostStatus.ACTIVE));
+} catch (e) {
+  print("Query failed: " + e);
+}
```

Comment:
In the Android and iOS docs, we don't show the blocking versions of the calls. I'd suggest something instead like:

```suggestion
Amplify.DataStore.query(Comment.classType, where: Post.STATUS.eq(PostStatus.ACTIVE))
    .asStream()
    .listen((result) => {
       // yooo, dude, sick.
    })
    .onError((error) => {
      // damn bruh.
    })
```

^ Note: syntax might be slightly off.

Reply:
I'm not sure if this is the best way to show the usage of the APIs. `.asStream().listen()` is very similar to programming paradigm of using `then()` which creates a callback hell for developers to maintain. `async/await` never blocks the main thread, it just schedules the rest of the function body to execute after the await fulfills.